### PR TITLE
@types/humanize-duration - Add digitReplacements to Options

### PR DESCRIPTION
**Pull request by Sagie501**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<[url here](https://www.npmjs.com/package/humanize-duration#digitreplacements)>>

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/humanize-duration/humanize-duration-tests.ts

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 809498a052ea0e7ac33d067a9fa7b99fe9609301
</p>

File: types/humanize-duration/index.d.ts [hunk 1]
```diff
@@ -140,6 +140,10 @@ declare namespace humanizeDuration {
          * Customize the value used to calculate each unit of time.
          */
         unitMeasures?: UnitMeasuresOptions | undefined;
+        /**
+         * Array of ten strings to which will replace the numerals 0-9.
+         */
+        digitReplacements?: string[] | undefined;
```
Comment (EvanHahn) [inline types/humanize-duration/index.d.ts:146]: This works but doesn't protect against arrays of length != 9. Consider:

```suggestion
        digitReplacements?: Record<"0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9", string> | undefined;
```

See [the JSDoc types in HumanizeDuration.js](https://github.com/EvanHahn/HumanizeDuration.js/blob/392c869c8f7db7f359ab9011e05a8f32a9e5cf66/humanize-duration.js#L26) for more.

**Review by EvanHahn — CHANGES_REQUESTED** [review]
Looks like things need to be auto-formatted. LGTM otherwise, thanks for doing this!

File: types/humanize-duration/index.d.ts (hunk 1 above)
Comment (Sagie501) [reply types/humanize-duration/index.d.ts:146]: [The docs](https://www.npmjs.com/package/humanize-duration#digitreplacements) are referring it as an array so specify it as `Record` might be a bit confusing IMO.

Maybe the docs should be updated or specify the type as:
```typescript
digitReplacements?: [string, string, string, string, string, string, string, string, string, string] | undefined;
```

WDYT?

Comment (EvanHahn) [reply types/humanize-duration/index.d.ts:146]: Good point. I've [updated HumanizeDuration's type][0] based on your suggestion. Let's go with that!

[0]: https://github.com/EvanHahn/HumanizeDuration.js/commit/83ff8534f1e7916a8bedb585a3bbfb23e1683327

Comment (Sagie501) [reply types/humanize-duration/index.d.ts]: Done :)

**Review by EvanHahn — APPROVED** [review]
Looks good. Thanks for this work!

**Comment by EvanHahn** [comment]
Ready to merge

**Pull request merged**